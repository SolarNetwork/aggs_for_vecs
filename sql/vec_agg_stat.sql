-- numeric stats
SELECT vec_agg_count(nums) AS counts
	, vec_agg_sum(nums) AS sums
	, vec_trim_scale(vec_agg_mean(nums)) AS means
	, vec_agg_min(nums) AS mins
	, vec_agg_max(nums) AS maxes
FROM measurements;

-- numeric stats 2
SELECT vec_agg_count(data_i) AS counts
	, vec_agg_sum(data_i) AS sums
	, vec_trim_scale(vec_agg_mean(data_i)) AS means
	, vec_agg_min(data_i) AS mins
	, vec_agg_max(data_i) AS maxes
FROM measurements2;