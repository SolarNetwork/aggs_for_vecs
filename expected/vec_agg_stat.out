-- int2 stats
SELECT vec_agg_count(smallints) AS counts
	, vec_agg_sum(smallints) AS sums
	, vec_agg_mean(smallints) AS means
	, vec_agg_min(smallints) AS mins
	, vec_agg_max(smallints) AS maxes
FROM measurements;
 counts  |   sums   |                             means                              |  mins   |  maxes  
---------+----------+----------------------------------------------------------------+---------+---------
 {4,3,4} | {4,5,13} | {1.00000000000000000000,1.6666666666666667,3.2500000000000000} | {1,1,2} | {1,2,4}
(1 row)

-- int4 stats
SELECT vec_agg_count(ints) AS counts
	, vec_agg_sum(ints) AS sums
	, vec_agg_mean(ints) AS means
	, vec_agg_min(ints) AS mins
	, vec_agg_max(ints) AS maxes
FROM measurements;
 counts  |   sums   |                             means                              |  mins   |  maxes  
---------+----------+----------------------------------------------------------------+---------+---------
 {4,3,4} | {4,5,13} | {1.00000000000000000000,1.6666666666666667,3.2500000000000000} | {1,1,2} | {1,2,4}
(1 row)

-- int8 stats
SELECT vec_agg_count(bigints) AS counts
	, vec_agg_sum(bigints) AS sums
	, vec_agg_mean(bigints) AS means
	, vec_agg_min(bigints) AS mins
	, vec_agg_max(bigints) AS maxes
FROM measurements;
 counts  |   sums   |                             means                              |  mins   |  maxes  
---------+----------+----------------------------------------------------------------+---------+---------
 {4,3,4} | {4,5,13} | {1.00000000000000000000,1.6666666666666667,3.2500000000000000} | {1,1,2} | {1,2,4}
(1 row)

-- float4 stats
SELECT vec_agg_count(reals) AS counts
	, vec_agg_sum(reals) AS sums
	, vec_agg_mean(reals) AS means
	, vec_agg_min(reals) AS mins
	, vec_agg_max(reals) AS maxes
FROM measurements;
 counts  |   sums   |            means            |  mins   |  maxes  
---------+----------+-----------------------------+---------+---------
 {4,3,4} | {4,5,13} | {1,1.6666666666666667,3.25} | {1,1,2} | {1,2,4}
(1 row)

-- float8 stats
SELECT vec_agg_count(floats) AS counts
	, vec_agg_sum(floats) AS sums
	, vec_agg_mean(floats) AS means
	, vec_agg_min(floats) AS mins
	, vec_agg_max(floats) AS maxes
FROM measurements;
 counts  |   sums   |            means            |  mins   |  maxes  
---------+----------+-----------------------------+---------+---------
 {4,3,4} | {4,5,13} | {1,1.6666666666666667,3.25} | {1,1,2} | {1,2,4}
(1 row)

-- numeric stats
SELECT vec_agg_count(nums) AS counts
	, vec_agg_sum(nums) AS sums
	, vec_trim_scale(vec_agg_mean(nums)) AS means
	, vec_agg_min(nums) AS mins
	, vec_agg_max(nums) AS maxes
FROM measurements;
 counts  |       sums        |      means      |       mins       |      maxes       
---------+-------------------+-----------------+------------------+------------------
 {4,3,4} | {4.92,5.91,14.80} | {1.23,1.97,3.7} | {1.23,1.23,2.34} | {1.23,2.34,4.56}
(1 row)

-- numeric stats 2
SELECT vec_agg_count(data_i) AS counts
	, vec_agg_sum(data_i) AS sums
	, vec_trim_scale(vec_agg_mean(data_i)) AS means
	, vec_agg_min(data_i) AS mins
	, vec_agg_max(data_i) AS maxes
FROM measurements2;
                counts                 |                                                sums                                                 |                                                                                 means                                                                                  |                                  mins                                  |                                     maxes                                     
---------------------------------------+-----------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------+-------------------------------------------------------------------------------
 {184,184,184,184,184,184,184,184,184} | {35968097,3460.738929,42273.93922,9198.984594,25360099,183.07443653,-971.9407094,46903731,25143722} | {195478.788043478261,18.8083637445652174,229.7496696739130435,49.9944814891304348,137826.625,0.99496976375,-5.2822864641304348,254911.58152173913,136650.663043478261} | {0,-75585.95336,225.00002,49.500015,2531,0.09494743,-332.28107,2531,0} | {23200619,48587.514387,234.99962,50.5,13161880,1,332.20728,23730754,23623545}
(1 row)

-- numeric stats 3
SELECT vec_agg_count(pad_vec(data_a, 3)) AS counts
	, vec_agg_sum(pad_vec(data_a, 3)) AS sums
	, vec_trim_scale(vec_agg_mean(pad_vec(data_a, 3))) AS means
	, vec_agg_min(pad_vec(data_a, 3)) AS mins
	, vec_agg_max(pad_vec(data_a, 3)) AS maxes
FROM measurements2;
   counts    |            sums             |               means                |         mins          |         maxes         
-------------+-----------------------------+------------------------------------+-----------------------+-----------------------
 {184,0,180} | {1164885081,NULL,11117.700} | {6330897.179347826087,NULL,61.765} | {6255484,NULL,61.765} | {6477419,NULL,61.765}
(1 row)

